overlord/snapstate/autorefresh_gating.go

```diff
@@ -220,3 +220,17 @@ func usesMountBackend(iface interfaces.Interface) bool {
 	}
 	return false
 }
+
+func createAutoRefreshGateHooks(st *state.State, affectedSnaps map[string]*affectedSnapInfo) *state.TaskSet {
+	ts := state.NewTaskSet()
+	var prev *state.Task
+	for snapName, affected := range affectedSnaps {
+		hookTask := SetupAutoRefreshGatingHook(st, snapName, affected.Base, affected.Restart)
+		if prev != nil {
+			hookTask.WaitFor(prev)
```

Comment:
this makes them run sequentially,  I suppose it's ok but is hard to judget here, anyway easy to adjust later but the function probably needs a doc comment that explain a bit more what it does and this

Reply:
Fair point, I think it will be fine to run the parallel. I'll revisit and maybe drop this when it's fleshed out.